fir/src/org/jetbrains/kotlin/idea/fir/intentions/HLAddNameToArgumentIntention.kt

```diff
@@ -0,0 +1,102 @@
+import org.jetbrains.kotlin.idea.frontend.api.calls.KtFunctionCall
+import org.jetbrains.kotlin.idea.frontend.api.calls.getSingleCandidateSymbolOrNull
+import org.jetbrains.kotlin.idea.project.languageVersionSettings
+import org.jetbrains.kotlin.idea.util.textRangeIn
+import org.jetbrains.kotlin.name.Name
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.getPrevSiblingIgnoringWhitespace
+
+class HLAddNameToArgumentIntention :
+    AbstractHLIntention<KtValueArgument, HLAddNameToArgumentIntention.Input>(KtValueArgument::class, applicator), LowPriorityAction {
+    class Input(val name: Name) : HLApplicatorInput
+
+    override val applicabilityRange: HLApplicabilityRange<KtValueArgument> = applicabilityRanges { element ->
+        val expression = element.getArgumentExpression() ?: return@applicabilityRanges emptyList()
+        if (expression is KtLambdaExpression) {
+            // Use OUTSIDE of curly braces only as applicability ranges for lambda inside an argument list (e.g., `run({  })`).
+            // If we use the text range of the curly brace elements, it will include the inside of the braces. This matches FE 1.0 behavior.
+            // Note: Intention is NOT applicable when lambda is trailing lambda after argument list (e.g., `run {  }`).
+            listOfNotNull(TextRange(0, 0), TextRange(element.textLength, element.textLength))
+        } else {
+            listOf(TextRange(0, element.textLength))
+        }
+    }
+
+    override val inputProvider: HLApplicatorInputProvider<KtValueArgument, Input> = inputProvider { element ->
+        val argumentList = element.parent as? KtValueArgumentList ?: return@inputProvider null
+        val shouldBeLastUnnamed = !element.languageVersionSettings.supportsFeature(LanguageFeature.MixedNamedArgumentsInTheirOwnPosition)
+        if (shouldBeLastUnnamed && element != argumentList.arguments.last { !it.isNamed() }) return@inputProvider null
+
+        // TODO: Cast to KtCallElement (need to handle other subtypes in KtCallResolver mixin)
+        val callElement = argumentList.parent as? KtCallExpression ?: return@inputProvider null
+        val resolvedCall = callElement.resolveCall() as? KtFunctionCall ?: return@inputProvider null
+
+        if (resolvedCall.targetFunction.getSingleCandidateSymbolOrNull()?.hasStableParameterNames != true) {
+            return@inputProvider null
+        }
+
+        val valueParameterSymbol = resolvedCall.argumentMapping[element] ?: return@inputProvider null
+        if (valueParameterSymbol.isVararg) {
+            if (element.languageVersionSettings.supportsFeature(LanguageFeature.ProhibitAssigningSingleElementsToVarargsInNamedForm) &&
+                !element.isSpread) {
+                return@inputProvider null
+            }
+
+            val varargArgumentCount = resolvedCall.argumentMapping.values.count { it == valueParameterSymbol }
+            if (varargArgumentCount != 1) {
```

Comment:
Are we going to check if this `vararg` appears only and exactly once?

Reply:
We can only add the parameter name for an argument for a vararg parameter if it's the ONLY argument for the parameter. E.g.,:
```kt
fun foo(vararg i: Int) {}

foo(1, 2) // Can NOT add `i = ` to either argument
foo(1) // Can change to `i = 1`
```
Adding a comment to the code.

Reply:
Maybe add those examples to code comments as well.

Reply:
Of course, I should have! Done.